File: src/main/java/cn/chahuyun/economy/entity/TitleInfo.java
package cn.chahuyun.economy.entity;

import cn.chahuyun.economy.constant.TitleTemplate;
import cn.chahuyun.economy.utils.ImageUtil;
import jakarta.persistence.*;
import lombok.*;

import java.awt.*;
import java.io.Serializable;
import java.util.Date;

/**
 * 称号信息<p>
 *
 * @author Moyuyanli
 * @date 2022/12/5 17:01
 */
@Getter
@Setter
@Entity(name = "TitleInfo")
@Table
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TitleInfo implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    /**
     * 所属者用户id
     */
    private long userId;
    /**
     *
     */
    private TitleTemplate type;
    /**
     * 使用状态
     */
    private boolean status;
    /**
     * 称号
     */
    private String title;
    /**
     * 是否影响名称
     */
    private boolean impactName;
    /**
     * 是否渐变
     */
    private boolean gradient;
    /**
     * 称号颜色
     */
    private String sColor;
    /**
     * 称号颜色
     */
    private String eColor;
    /**
     * 称号到期时间
     */
    private Date dueTime;

    public Color getStartColor() {
        return ImageUtil.hexColor(sColor);
    }

    public Color getEndColor() {
        return ImageUtil.hexColor(eColor);
    }

}

File: src/main/kotlin/cn/chahuyun/economy/entity/TitleInfo.kt
package cn.chahuyun.economy.entity

import cn.chahuyun.economy.utils.ImageUtil
import jakarta.persistence.*
import java.awt.Color
import java.io.Serializable
import java.util.*

/**
 * 称号信息
 *
 * @author Moyuyanli
 * @date 2022/12/5 17:01
 */
@Entity(name = "TitleInfo")
@Table
class TitleInfo(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Int = 0,

    /**
     * 所属者用户id
     */
    var userId: Long = 0,

    /**
     * 称号类型
     */
    var code: String? = null,

    /**
     * 称号名称
     */
    var name: String? = null,

    /**
     * 使用状态
     */
    var status: Boolean = false,

    /**
     * 称号
     */
    var title: String? = null,

    /**
     * 是否影响名称
     */
    var impactName: Boolean = false,

    /**
     * 是否渐变
     */
    var gradient: Boolean = false,

    /**
     * 称号颜色
     */
    var sColor: String? = null,

    /**
     * 称号颜色
     */
    var eColor: String? = null,

    /**
     * 称号到期时间
     */
    var dueTime: Date? = null
) : Serializable {

    val startColor: Color
        get() = ImageUtil.hexColor(sColor)

    val endColor: Color
        get() = ImageUtil.hexColor(eColor)

    constructor(userId: Long, code: String?, name: String?, title: String?) : this(
        userId = userId,
        code = code,
        name = name,
        title = title,
        status = false
    )
}
